android/src/com/frostwire/android/gui/fragments/ImageViewerFragment.java

```diff
@@ -0,0 +1,194 @@
+package com.frostwire.android.gui.fragments;
+
+import android.app.Activity;
+import android.net.Uri;
+import android.os.SystemClock;
+import android.support.v7.view.ActionMode;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+
+import com.frostwire.android.AndroidPlatform;
+import com.frostwire.android.R;
+import com.frostwire.android.core.FileDescriptor;
+import com.frostwire.android.gui.adapters.menu.DeleteFileMenuAction;
+import com.frostwire.android.gui.adapters.menu.OpenMenuAction;
+import com.frostwire.android.gui.adapters.menu.RenameFileMenuAction;
+import com.frostwire.android.gui.adapters.menu.SeedAction;
+import com.frostwire.android.gui.adapters.menu.SendFileMenuAction;
+import com.frostwire.android.gui.adapters.menu.SetAsWallpaperMenuAction;
+import com.frostwire.android.gui.util.UIUtils;
+import com.frostwire.android.gui.views.AbstractDialog;
+import com.frostwire.android.gui.views.AbstractFragment;
+import com.frostwire.android.util.ImageLoader;
+
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ImageViewerFragment extends AbstractFragment {
+    //private static final Logger LOG = Logger.getLogger(ImageViewerFragment.class);
+    private ImageView imageView;
+    private FileDescriptor fd;
+    private ImageViewerActionModeCallback actionModeCallback;
+
+    public ImageViewerFragment() {
+        super(R.layout.fragment_image_viewer);
+        setHasOptionsMenu(true);
+        actionModeCallback = null;
+    }
+
+    @Override
+    protected void initComponents(View v) {
+        imageView = findView(v, R.id.fragment_image_viewer_image);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (fd != null) {
```

Comment:
I can remove this, when it resumes the `fd` is still there somehow, was just being cautious